src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -479,948 +575,1525 @@ public void LastIndexOf()
         {
-            Assert.Equal(s_empty, s_empty.AddRange(s_empty));
-            Assert.Equal(s_oneElement, s_empty.AddRange(s_oneElement)); // struct overload
-            Assert.Equal(s_oneElement, s_empty.AddRange((IEnumerable<int>)s_oneElement)); // enumerable overload
-            Assert.Equal(s_oneElement, s_oneElement.AddRange(s_empty));
+            yield return new object[] { s_twoElementRefTypeWithNull, string.Empty };
+            yield return new object[] { new[] { default(object) }, new object() };
+            yield return new object[] { new[] { null, new object() }, new object() };
+            yield return new object[] { new[] { null, string.Empty }, string.Empty };
         }
 
-        [Fact]
-        public void Insert()
+        [Theory]
+        [MemberData(nameof(EqualsData))]
+        public void Equals(StrongBox<ImmutableArray<int>> firstBox, StrongBox<ImmutableArray<int>> secondBox, bool expected)
         {
-            var array1 = ImmutableArray.Create<char>();
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => array1.Insert(-1, 'a'));
-            Assert.Throws<ArgumentOutOfRangeException>("index", () => array1.Insert(1, 'a'));
+            ImmutableArray<int> first = firstBox.Value;
+            ImmutableArray<int> second = secondBox.Value;
 
-            var insertFirst = array1.Insert(0, 'c');
-            Assert.Equal(new[] { 'c' }, insertFirst);
+            Assert.Equal(expected, first == second);
+            Assert.NotEqual(expected, first != second);
 
-            var insertLeft = insertFirst.Insert(0, 'a');
-            Assert.Equal(new[] { 'a', 'c' }, insertLeft);
+            Assert.Equal(expected, first.Equals(second));
+            Assert.Equal(expected, AsEquatable(first).Equals(second));
+            Assert.Equal(expected, first.Equals((object)second));
 
-            var insertRight = insertFirst.Insert(1, 'e');
-            Assert.Equal(new[] { 'c', 'e' }, insertRight);
+            Assert.Equal(expected, second == first);
+            Assert.NotEqual(expected, second != first);
 
-            var insertBetween = insertLeft.Insert(1, 'b');
-            Assert.Equal(new[] { 'a', 'b', 'c' }, insertBetween);
+            Assert.Equal(expected, second.Equals(first));
+            Assert.Equal(expected, AsEquatable(second).Equals(first));
+            Assert.Equal(expected, second.Equals((object)first));
         }
 
-        [Fact]
-        public void InsertDefault()
+        public static IEnumerable<object[]> EqualsData()
         {
-            Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(-1, 10));
-            Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(1, 10));
-            Assert.Throws<NullReferenceException>(() => s_emptyDefault.Insert(0, 10));
-        }
+            // Once https://github.com/xunit/assert.xunit/pull/5 comes into corefx, all the StrongBox stuff can be removed.
```

Comment:
Instead of doing this workaround, why don't we bring those necessary changes into buildtools?